hooks/tk-multi-publish2/basic/collector.py

```diff
@@ -0,0 +1,330 @@
+# Copyright (c) 2017 Shotgun Software Inc.
+# 
+# CONFIDENTIAL AND PROPRIETARY
+# 
+# This work is provided "AS IS" and subject to the Shotgun Pipeline Toolkit 
+# Source Code License included in this distribution package. See LICENSE.
+# By accessing, using, copying or modifying this work you indicate your 
+# agreement to the Shotgun Pipeline Toolkit Source Code License. All rights 
+# not expressly granted therein are reserved by Shotgun Software Inc.
+
+import glob
+import os
+import maya.cmds as cmds
+import maya.mel as mel
+import sgtk
+
+HookBaseClass = sgtk.get_hook_baseclass()
+
+
+class MayaSessionCollector(HookBaseClass):
+    """
+    Collector that operates on the maya session. Should inherit from the basic
+    collector hook.
+    """
+
+    @property
+    def settings(self):
+        """
+        Dictionary defining the settings that this collector expects to receive
+        through the settings parameter in the process_current_session and
+        process_file methods.
+
+        A dictionary on the following form::
+
+            {
+                "Settings Name": {
+                    "type": "settings_type",
+                    "default": "default_value",
+                    "description": "One line description of the setting"
+            }
+
+        The type string should be one of the data types that toolkit accepts as
+        part of its environment configuration.
+        """
+
+        # grab any base class settings
+        collector_settings = super(MayaSessionCollector, self).settings or {}
```

Comment:
Because we only have a single collector, we need to add the subclass settings to the base class offering.